### PR TITLE
translation(js): /Web/JavaScript/Reference/Template_literals

### PR DESCRIPTION
**Pull request by AdriandeCita**
Original content: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

### CODE REVIEW

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+  `icon-${item.isCollapsed ? 'expander' : 'collapser'}` }`;
+```
+
+### Теговані шаблони
+
+Більш розширена форма шаблонних літералів — це _теговані_ шаблони.
+
+Теги дають змогу розбирати шаблонні літерали за допомогою функції. Перший аргумент тегової функції містить масив рядкових значень. Решта аргументів належать до виразів.
+
+Тегова функція далі може виконати які завгодно операції на цих аргументах, і повернути оброблений рядок. (Вона також може повертати щось абсолютно інше, як описано в одному з прикладів нижче.)
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:140]:
```suggestion
Тегова функція далі може виконати які завгодно операції на цих аргументах і повернути оброблений рядок. (Вона також може повертати щось абсолютно інше, як описано в одному з прикладів нижче.)
```

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+function tag(strings) {
+  console.log(strings.raw[0]);
+}
+
+tag`стрічка тексту — рядок 1 \n стрічка тексту — рядок 2`;
+// друкує "стрічка тексту — рядок 1 \n стрічка тексту — рядок 2" ,
+// включно з двома символами '\' та 'n'
+```
+
+На додаток, для створення необроблених рядків існує метод {{jsxref("String.raw()")}}, який працює точнісінько так само, як усталена тегова функція, і конкатенація рядків.
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:213]:
```suggestion
На додаток, для створення необроблених рядків існує метод {{jsxref("String.raw()")}}, який працює точнісінько так само, як усталена тегова функція і конкатенація рядків.
```

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
+Array.from(str).join(',');
+// "П,р,и,в,і,т,\\,n,5,!"
+```
+
+### Теговані шаблони і екрановані послідовності
+
+#### Поведінка згідно ES2016
+
+Від ECMAScript 2016, теговані шаблони враховують правила таких екранованих послідовностей:
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:230]:
```suggestion
Починаючи з ECMAScript 2016, теговані шаблони враховують правила таких екранованих послідовностей:
```

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+- Екрановані послідовності Unicode, що починаються з "`\u`", наприклад `\u00A9`
+- Екрановані послідовності кодової точки Unicode, що позначаються "`\u{}`", наприклад `\u{2F804}`
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:233]:
```suggestion
- Екрановані послідовності кодової точки Unicode, що позначаються "`\u{}`", наприклад, `\u{2F804}`
```

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+- Шістнадцяткові екранування, що починаються з "`\x`", наприклад `\xA9`
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:234]:
```suggestion
- Шістнадцяткові екранування, що починаються з "`\x`", наприклад, `\xA9`
```

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+- Вісімкові літеральні екранування, що починаються з "`\0o`", а продовжуються однією чи декількома цифрами, наприклад `\0o251`
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:235]:
```suggestion
- Вісімкові літеральні екранування, що починаються з "`\0o`", а продовжуються однією чи декількома цифрами, наприклад, `\0o251`
```

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+
+Це означає, що буде проблематичним вжити таки тегований шаблон, як наведено нижче, оскільки, згідно з граматикою ECMAScript, парсер шукатиме дійсну екрановану послідовність Unicode, але знайде дефектний синтаксис:
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:237]:
```suggestion
Це означає, що буде проблематичним вжити такий тегований шаблон, як наведено нижче, оскільки, згідно з граматикою ECMAScript, парсер шукатиме дійсну екрановану послідовність Unicode, але знайде дефектний синтаксис:
```

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+```js
+latex`\unicode`
+// Викидає помилку в старіших версіях ECMAScript (ES2016 і раніших)
+// SyntaxError: malformed Unicode character escape sequence
+```
+
+#### Перегляд недійсних екранованих послідовностей в ES2018
+
+Теговані шаблони повинні давати змогу вкладати інші мови (наприклад [предметно орієнтовані](https://en.wikipedia.org/wiki/Domain-specific_language), чи [LaTeX](https://en.wikipedia.org/wiki/LaTeX)), в яких поширені інші екрановані послідовності. Пропозиція до розгляду ECMAScript під назвою [Перегляд шаблонних літералів (англ.)](https://tc39.github.io/proposal-template-literal-revision/) (інтегровано в стандарт ECMAScript 2018) прибрала обмеження екранованих послідовностей ECMAScript з тегованих шаблонів.
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:247]:
```suggestion
Теговані шаблони повинні давати змогу вкладати інші мови (наприклад, [предметно орієнтовані](https://uk.wikipedia.org/wiki/%D0%9F%D1%80%D0%B5%D0%B4%D0%BC%D0%B5%D1%82%D0%BD%D0%BE-%D0%BE%D1%80%D1%96%D1%94%D0%BD%D1%82%D0%BE%D0%B2%D0%B0%D0%BD%D0%B0_%D0%BC%D0%BE%D0%B2%D0%B0_%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D1%83%D0%B2%D0%B0%D0%BD%D0%BD%D1%8F), чи [LaTeX](https://uk.wikipedia.org/wiki/LaTeX)), в яких поширені інші екрановані послідовності. Пропозиція до розгляду ECMAScript під назвою [Перегляд шаблонних літералів (англ.)](https://tc39.github.io/proposal-template-literal-revision/) (інтегровано в стандарт ECMAScript 2018) прибрала обмеження екранованих послідовностей ECMAScript з тегованих шаблонів.
```

File: files/uk/web/javascript/reference/template_literals/index.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+
+Все ж, недійсні екрановані послідовності все одно необхідно показати в "приготовленому" виразі. Їх буде показано як {{jsxref("undefined", "невизначені")}} елементи в масиві "cooked":
```
Comment (undead404) [inline files/uk/web/javascript/reference/template_literals/index.md:249]:
```suggestion
Все ж, недійсні екрановані послідовності все одно необхідно показати в "приготованому" виразі. Їх буде показано як {{jsxref("undefined", "невизначені")}} елементи в масиві "cooked":
```

**Review by undead404 — CHANGES_REQUESTED** [review]

**Comment by AdriandeCita** [comment]
Оновлено, додано нові зміни: [7449143](https://github.com/mdn/content/commit/74491431d2afdcb4049efe037349361dca9329b6), [461710a](https://github.com/mdn/content/commit/461710a5d168c58ca67818895059772a26a17de3), [dd242c9](https://github.com/mdn/content/commit/dd242c9a48f7c34d068fb10bbe3f4f0fd34a17bd), [6fb517d](https://github.com/mdn/content/commit/6fb517d5f9724399d490fb9cda7a03898f40e2fd), [76a015f](https://github.com/mdn/content/commit/76a015ffc44bd29c93e725366ad023171e6c7e7d), [37676df](https://github.com/mdn/content/commit/37676df66bb7cf9c7b59ac573c98febe401c9eed), [c502c81](https://github.com/mdn/content/commit/c502c811a563440296bfd29d1d9862772005d337), [a94ef2c](https://github.com/mdn/content/commit/a94ef2c598fc7b6a8006bb66227da870df61e3d6), [529663b](https://github.com/mdn/content/commit/529663b77721cded2c6ea2634d89d0fc38481bcb), [4794d70](https://github.com/mdn/content/commit/4794d70196f15bff02cbd8334665eaed61959ac8)

**Review by undead404 — APPROVED** [review]

**Pull request merged**